### PR TITLE
config: improve port allocation + cleanup

### PR DESCRIPTION
**Pull request by bmwill**
The bulk of this PR is around the better allocation of ports:

    config: better allocation of available ports
    
    Its not ideal to attempt to allocate ports for a large number of nodes
    up front. Generally the better thing to do would be to always bind to
    port 0 and allow from the OS to properly allocate an available port and
    then communicate the exact port that a network interface is bound to to
    other entities that would like to speak with it inside of a test
    environment.
    
    Given the above still requires a bit of work and plumbing to be done the
    next best thing is to be a bit more intelligent about how we're
    allocating ports when constructing test configs. This patch changes how
    ports are allocated by requesting a fresh, available port from the OS by
    binding on port 0, immediately establishing a connection, and then
    dropping the connection and the listener. This forces the OS to put the
    port into the TIME_WAIT state ensuring that the OS won't hand out the
    port for some grace period. Then the port can later be used by by
    binding on the port using SO_REUSEADDR (which most rust networking
    libraries do by default).

### CODE REVIEW

File: sui/src/config/mod.rs [hunk 1]
```diff
@@ -431,15 +439,12 @@ pub fn make_default_narwhal_committee(
 ) -> Result<ConsensusCommittee<Ed25519PublicKey>, anyhow::Error> {
     let mut ports = Vec::new();
     for _ in authorities {
-        let mut authority_ports = Vec::new();
```
Comment (velvia) [inline sui/src/config/mod.rs:434]: Say can you add a better RustDoc to this method, that it should only be for testing/

**Review by velvia — APPROVED** [review]
Hey @bmwill this looks fine, but is it possible to leave out the 90% of the review which is just import refactoring?  I don't know if that's your IDE but unless rustfmt settings changed, can we leave that out because it makes reviewing it much more noisy.  Thanks

File: sui/src/config/mod.rs (hunk 1 above)
Comment (bmwill) [reply sui/src/config/mod.rs:434]: I don't know the intent of this function, I'll let @asonnino comment on it's intent since it's narwhal related

**Comment by bmwill** [comment]
I generally break my PRs up into atomic commits so you should be able to look at the one commit that does the port stuff and reduce the noise.

And no all of the other stuff was intentional as I was also moving around how/where some items are exported.

**Pull request merged**